router/batchrouter/asyncdestinationmanager/eloqua/eloqua.go

```diff
@@ -0,0 +1,223 @@
+package eloqua
```

Comment:
[Minor]
In this file, I'm seeing the part from requesting data to getting the response is more or less similar
```golang
type ReqParams struct {
 endpoint string
 requestMethod string
 body <typeOf Strings.NewReader>
 authorisation string
 contentType string
}

func makeHttpRequest(reqParams ReqParams) (<typeOf body>, error) {
        req, err := http.NewRequest(reqParams.requestMethod, reqParams.endpoint, reqParams.body)

        if (reqParams.authorisation != "") {
	   req.Header.Add("Authorization", reqParams.authorisation)
        }
        if (reqParams.contentType != "") {
	   req.Header.Add("Content-Type", reqParams.contentType)
        }
	if err != nil {
		return nil, err
	}
	res, err := http.DefaultClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer res.Body.Close()
	body, err := io.ReadAll(res.Body)
	if err != nil {
		return nil, err
	}
        return body
}
// where ever we want to use we can do 
makeHttpRequest(ReqParams{...})
```